README.md

```diff
@@ -0,0 +1,198 @@
+
+``` bash
+# run test specified with a pattern or filename
+yarn jest my-test #or
+yarn jest path/to/my-test.js
+
+# run tests related to changed files based on hg/git (uncommitted files)
+yarn jest -o
+
+# run tests related to path/to/fileA.js and path/to/fileB.js
+yarn jest --findRelatedTests path/to/fileA.js path/to/fileB.js
+
+# run tests that match this spec name (match against the name in describe or test, basically).
+yarn jest -t name-of-spec
+
+# run tests in watch mode
+yarn jest --watch #runs jest -o by default
+yarn jest --watchAll #runs all tests
+```
+
+### Build
+
+#### Production
+
+To build the application run: `yarn build`, the build files are written to `dist` directory. To run the application using the generated build execute `yarn electron dist/main`
+
+`yarn build` is composed by two sub-tasks:
+
+* `yarn build-renderer`: Build the frontend/ui (React) application that will run in Electron's renderer process
+* `yarn build-main`: Build the backend (Node) application that will run in Electron's main process
+
+#### Development
+
+To build the application run: `yarn build-dev`, the build files are written to `dist` directory. To run the application using the generated build execute `yarn electron dist/main`. Note that in development mode, the application needs a running development server that serves the static assets, this means that you need to run `yarn hot-server` in another terminal in order to view the application with all the assets.
+
+#### Type Checking
+
+We have separated the transpilation and the type-checking phase for development in order to reduce the time the application needs to boot in development mode. Building/running the application for development with `yarn build-dev`/`yarn start` won't do any type-checking, the same happens when running the tests with `yarn test`.
+
+To type-check the project you should run:
+
+``` bash
+yarn typecheck
+# or
+# yarn typecheck --watch
+```
+
+To type-check the project and the tests you should run:
+
+``` bash
+yarn typecheck-test
+# or
+# yarn typecheck-test --watch
+```
+
+### Package
+
+To package the application into an executable for your platform (Windows, MacOS, GNU/Linux) run any of the following commands. The packaged application will be written to `release` folder.
+
+#### Production build
+
+``` bash
+yarn package
+```
+
+#### Development build
+
+``` bash
+yarn package-dev
+```
+
+### Contributing
+
+#### Commit Messages
+
+Use the command `yarn commit` when committing changes to follow the same conventions we used for writing commit messages.
+
+#### Travis (continuous integration)
+
+When opening a pull request a job in [Travis](https://travis-ci.com/) will be fired off to check the changes. To avoid wasting time waiting for Travis output we provide the command `yarn travis` that will perform almost the same checks but it'll run in your computer. The command `yarn travisp` does the same but parallelizes some tasks so it'll run faster.
```

Comment:
```suggestion
When opening a pull request a job in [Travis](https://travis-ci.com/) will be fired off to check the changes. To avoid wasting time waiting for Travis output we provide the command `yarn ci` that will perform almost the same checks but it'll run in your computer.
```